build-logic/build-platform/build.gradle.kts

```diff
@@ -6,7 +6,7 @@ description = "Provides a platform that constrains versions of external dependen
 
 // Here you should declare versions which should be shared by the different modules of buildSrc itself
 val javaParserVersion = "3.18.0"
-val groovyVersion = "3.0.9"
+val groovyVersion = "4.0.0-beta-1"
```

Comment:
Why not 4.0.0?

Reply:
No, there are two bugs that prevent us from upgrading even to 4.0.0-beta-2:

https://issues.apache.org/jira/browse/GROOVY-10466
https://issues.apache.org/jira/browse/GROOVY-10467